src/Searchbar.vue

```diff
@@ -1,55 +1,84 @@
 <script>
 import typeahead from './Typeahead.vue';
+import searchbarPageItem from './SearchbarPageItem.vue';
 
 export default {
   extends: typeahead,
   computed: {
     primitiveData() {
-      function getTotalMatches(searchTarget, regexes) {
-        return regexes.reduce((total, regex) => (regex.test(searchTarget) ? total + 1 : total), 0);
+      // Returns the total number of matches between an array of regex patterns and string search targets.
+      function getTotalMatches(regexes, ...searchTargets) {
+        const searchTarget = searchTargets.join(' ');
+
+        return regexes.reduce((total, regex) => {
+          const matches = searchTarget.match(regex);
+          return total + (matches ? matches.length : 0);
+        }, 0);
       }
 
-      if (this.value.length < 2) {
+      if (this.value.length < 2 || !this.data) {
         return [];
       }
-      if (!this.data) {
-        return undefined;
-      }
-      const matches = [];
+      const pages = [];
       const regexes = this.value.split(' ')
         .filter(searchKeyword => searchKeyword !== '')
         .map(searchKeyword => searchKeyword.replace(/[.*+?^${}()|[\]\\]/g, '\\$&'))
-        .map(searchKeyword => new RegExp(searchKeyword, 'i'));
+        .map(searchKeyword => new RegExp(searchKeyword, 'ig'));
       this.data.forEach((entry) => {
-        const { headings, src, title } = entry;
+        const {
+          headings,
+          src,
+          title,
+          headingKeywords,
+        } = entry;
         const keywords = entry.keywords || '';
-        let searchTarget = [title].concat(keywords).concat(Object.values(headings)).join(' ');
-        let totalMatches = getTotalMatches(searchTarget, regexes);
-        if (totalMatches > 0) {
-          searchTarget = [title].concat(keywords).join(' ');
-          const isMatchingPage = getTotalMatches(searchTarget, regexes) === totalMatches;
-          if (isMatchingPage) {
-            matches.push(Object.assign(entry, { totalMatches }));
-          }
+        const displayTitle = title || src;
+
+        // The total number of occurrences of all indexed words ( headings, keywords, title ) for a page
```

Comment:
Just a nit, but should we stick to the usual spacing conventions here for the bracket? (i.e. `(headings, keywords, title)`)